practices/follow-functional-core-imperative-shell.md

```diff
@@ -3,117 +3,153 @@
 
 Values serve as the boundary between the layers.
 The imperative shell should communicate with the functional core by passing value objects exclusively, avoiding objects or functions that could potentially induce side effects.
 This ensures that the functional core remains isolated from external state changes, promoting clarity, predictability, and maintainability in the codebase while facilitating easier testing, debugging, and refactoring.
 
-### Learning Curve
+### Potentially Steep Learning Curve
 
-Transitioning to the Functional Core, Imperative Shell practice may present a learning curve.
+Transitioning to the Functional Core, Imperative Shell practice may present a steep learning curve for teams.
 To facilitate this transition smoothly, it's recommended that developers with more knowledge mentor other developers through pair programming sessions.
 Additionally, fostering an environment of knowledge sharing, providing resources, and allocating time for developers to study the practice can greatly aid in its adoption and understanding across the team.
 
 ### Is Incompatible with Object-Oriented Programming
 
-The Functional Core, Imperative Shell practice is not mutually exclusive with Object-Oriented Programming (OOP) principles.
-As long as the functional core remains free of side effects, it can coexist within an object-oriented architecture.
-The Functional Core's logic could be encapsulated within methods of a class, ensuring that these methods return new objects instead of mutating their fields.
-The Functional Core can take advantage of polymorphism by receiving or returning objects of a class hierarchy with multiple subclasses to represent various inputs or outputs.
-Developers can implement the functional core while adhering to OOP principles and have the advantages of both paradigms.
+<ed> I'm having trouble understanding the pitfall/challenge/limitation with this one. The text below implies developers can take advantage of both paradigms. Should we rephrase this subhead? Seems like "Can Be Compatible with OOP" is more in line with the text below. </ed> 
+
+The Functional Core, Imperative Shell practice and object-oriented programming (OOP) principles are not mutually exclusive.
+As long as the functional core remains free of side effects, it can co-exist within an object-oriented architecture.
+The functional core's logic could be encapsulated within methods of a class, ensuring that these methods return new objects instead of mutating their fields.
+The functional core can take advantage of polymorphism by receiving or returning objects of a class hierarchy with multiple subclasses to represent various inputs or outputs. <ed> I'm assuming your target reader will follow along with this explanation just fine. But this is one example of where this practice reads (to me) much more technical than others. What might help is relating this nuance back to code maintainability. </ed> 
```

Comment:
I think this can/should be re-worked. Let me take a stab at re-writing this. I hesitate to relate it back to code maintainability here b/c in theory any practice can support multiple capabilities. If you look at the bottom of this file, you'll see that it supports both code maintainability and test automation.

Reply:
My attempted re-write:

The purpose of an object is to bundle behavior and state into a single, cohesive unit. In contrast, the functional core/imperative shell pattern aims to separate pure logic from state management.

When using the functional core/imperative shell pattern in an object-oriented application, the idea is to extract functional-style components that are then utilized by objects managing state and interacting with external systems. This approach works best when there are clear boundaries between pure computations and operations that cause side effects. However, avoid applying this pattern to objects where state and behavior are tightly intertwined, as it can add unnecessary complexity. The goal is to separate concerns while preserving the natural behavior of objects and maintaining the integrity of stateful components.

Reply:
Nicely done. I've reviewed, altered some line breaks, and added to Oct. 30 commit.